numbert/utils/utils_numbert.py

```diff
@@ -316,6 +324,158 @@ def _create_examples(self, data, set_type):
                     InputExample(guid=guid, text_a=text_a, text_b=text_b, label=str(labels[doc_ind])))
         return examples
 
+class TreccarProcessor(DataProcessor):
+    """Processor for the TREC-CAR data set."""
+
+    def load_qrels(self, path):
+        """Loads qrels into a dict of key: query_id, value: list of relevant doc ids."""
+        qrels = collections.defaultdict(set)
+        with open(path) as f:
+            for i, line in enumerate(f):
+                query_id, _, doc_id, relevance = line.rstrip().split(' ')
+                if int(relevance) >= 1:
+                    qrels[query_id].add(doc_id)
+                if i % 1000000 == 0:
+                    print('Loading qrels {}'.format(i))
+        return qrels
+
+    def load_run(self, path):
+        """Loads run into a dict of key: query_id, value: list of candidate doc ids."""
+
+        # We want to preserve the order of runs so we can pair the run file with the
+        # TFRecord file.
+        run = collections.OrderedDict()
+        with open(path) as f:
+            for i, line in enumerate(f):
+                query_id, _, doc_title, rank, _, _ = line.split(' ')
+                if query_id not in run:
+                    run[query_id] = []
+                run[query_id].append((doc_title, int(rank)))
+                if i % 1000000 == 0:
+                    print('Loading run {}'.format(i))
+        # Sort candidate docs by rank.
+        sorted_run = collections.OrderedDict()
+        for query_id, doc_titles_ranks in run.items():
+            sorted(doc_titles_ranks, key=lambda x: x[1])
+            doc_titles = [doc_titles for doc_titles, _ in doc_titles_ranks]
+            sorted_run[query_id] = doc_titles
+
+        return sorted_run
+
+    def load_collection(self, data_dir):
+        """Loads TREC-CAR's paraghaphs into a dict of key: title, value: paragraph."""
+        self.corpus = {}
+
+        start_time = time.time()
+        APPROX_TOTAL_PARAGRAPHS = 30000000
+        with open(os.path.join(data_dir, "paragraphCorpus/dedup.articles-paragraphs.cbor"), 'rb') as f:
+            for i, p in enumerate(trec_car_classes.iter_paragraphs(f)):
```

Comment:
Maybe use tqdm here? It will avoid the if below

Reply:
Much cleaner yes, I fixed it!